tests/perf/test_sheriffing_criteria/test_criteria_tracker.py

```diff
@@ -0,0 +1,288 @@
+import time
+from contextlib import contextmanager
+from datetime import timedelta
+from unittest.mock import MagicMock
+from dateutil.parser import parse as dateutil_parse
+from multiprocessing.pool import AsyncResult
+
+import pytest
+from freezegun import freeze_time
+
+from tests.perf.test_sheriffing_criteria.conftest import CASSETTES_RECORDING_DATE
+from treeherder.perf.exceptions import NoFiledBugs
+from treeherder.perf.sheriffing_criteria import (
+    CriteriaTracker,
+    EngineerTractionFormula,
+    FixRatioFormula,
+    RecordComputer,
+    CriteriaRecord,
+)
+from treeherder.perf.sheriffing_criteria.criteria_tracking import ResultsChecker
+from treeherder.utils import PROJECT_ROOT
+
+pytestmark = [pytest.mark.freeze_time(CASSETTES_RECORDING_DATE, tick=True)]
+
+RECORD_TEST_PATH = (PROJECT_ROOT / 'tests/sample_data/criteria-records.csv').resolve()
+EXPECTED_LAST_UPDATE = dateutil_parse(CASSETTES_RECORDING_DATE)
+EXPECTED_VALUE = 0.5
+TESTS_WITH_NO_DATA = [
+    ('awsy', 'Base Content Explicit'),
+    ('browsertime', 'allrecipes-cold'),
+    ('raptor', 'os-baseline-power'),
+    ('talos', 'a11yr'),
+]
+TESTS_WITH_EXPIRED_DATA = [
+    ('awsy', 'Base Content Heap Unclassified'),
+    ('browsertime', 'amazon'),
+    ('build_metrics', 'compiler warnings'),
+    ('raptor', 'raptor-ares6-firefox'),
+    ('talos', 'about_newtab_with_snippets'),
+]
+TESTS_WITH_UPDATED_DATA = [
+    ('awsy', 'Base Content JS'),
+    ('browsertime', 'amazon-cold'),
+    ('build_metrics', 'installer size'),
+    ('raptor', 'raptor-assorted-dom-firefox'),
+    ('talos', 'about_preferences_basic'),
+]
+recording_date = dateutil_parse(CASSETTES_RECORDING_DATE).isoformat()
+RECORDS_WITH_NO_DATA = [
+    CriteriaRecord(
+        Framework=test[0],
+        Suite=test[1],
+        EngineerTraction='',
+        FixRatio='',
+        LastUpdatedOn='',
+        ShouldSync='',
```

Comment:
There should be a test for the case when `ShouldSync` is False. Sorry if there is one and I missed it.